lib/mongoid/matchable/elem_match.rb

```diff
@@ -0,0 +1,37 @@
+# encoding: utf-8
+module Mongoid
+  module Matchable
+
+    class ElemMatch < Default
+
+      # Return true if a given predicate matches a sub document entirely
+      #
+      # @example Do the values match?
+      #   matcher.matches?({"$elemMatch" => {"a" => 1, "b" => 2}})
+      #
+      # @param [ Hash ] value The values to check.
+      #
+      # @return [ true, false ] If the values match.
+      def matches?(value)
+        if !@attribute.is_a?(Array) || !value.kind_of?(Hash) || !value["$elemMatch"].kind_of?(Hash)
+          return false
+        end
+
+        @attribute.each do |sub_document|
```

Comment:
Can we just change this to @attribute.all? as well as the line below (line 22)? We would need the explicit returns in that case.